
from PyQt5 import QtCore, QtGui, QtWidgets


class Ui_Calc(object):
    def setupUi(self, MainWindow):
        MainWindow.setObjectName("MainWindow")
        MainWindow.resize(331, 471)
        self.centralwidget = QtWidgets.QWidget(MainWindow)
        self.centralwidget.setObjectName("centralwidget")
        self.label = QtWidgets.QLabel(self.centralwidget)
        self.label.setGeometry(QtCore.QRect(0, 0, 331, 141))
        font = QtGui.QFont()
        font.setPointSize(15)
        self.label.setFont(font)
        self.label.setMidLineWidth(0)
        self.label.setObjectName("label")
        self.btn_1 = QtWidgets.QPushButton(self.centralwidget)
        self.btn_1.setGeometry(QtCore.QRect(0, 150, 91, 81))
        font = QtGui.QFont()
        font.setPointSize(15)
        self.btn_1.setFont(font)
        self.btn_1.setObjectName("btn_1")
        self.btn_2 = QtWidgets.QPushButton(self.centralwidget)
        self.btn_2.setGeometry(QtCore.QRect(90, 150, 91, 81))
        font = QtGui.QFont()
        font.setPointSize(15)
        self.btn_2.setFont(font)
        self.btn_2.setObjectName("btn_2")
        self.btn_3 = QtWidgets.QPushButton(self.centralwidget)
        self.btn_3.setGeometry(QtCore.QRect(180, 150, 91, 81))
        font = QtGui.QFont()
        font.setPointSize(15)
        self.btn_3.setFont(font)
        self.btn_3.setObjectName("btn_3")
        self.btn_4 = QtWidgets.QPushButton(self.centralwidget)
        self.btn_4.setGeometry(QtCore.QRect(0, 230, 91, 81))
        font = QtGui.QFont()
        font.setPointSize(15)
        self.btn_4.setFont(font)
        self.btn_4.setObjectName("btn_4")
        self.btn_5 = QtWidgets.QPushButton(self.centralwidget)
        self.btn_5.setGeometry(QtCore.QRect(90, 230, 91, 81))
        font = QtGui.QFont()
        font.setPointSize(15)
        self.btn_5.setFont(font)
        self.btn_5.setObjectName("btn_5")
        self.btn_6 = QtWidgets.QPushButton(self.centralwidget)
        self.btn_6.setGeometry(QtCore.QRect(180, 230, 91, 81))
        font = QtGui.QFont()
        font.setPointSize(15)
        self.btn_6.setFont(font)
        self.btn_6.setObjectName("btn_6")
        self.btn_7 = QtWidgets.QPushButton(self.centralwidget)
        self.btn_7.setGeometry(QtCore.QRect(0, 310, 91, 81))
        font = QtGui.QFont()
        font.setPointSize(15)
        self.btn_7.setFont(font)
        self.btn_7.setObjectName("btn_7")
        self.btn_8 = QtWidgets.QPushButton(self.centralwidget)
        self.btn_8.setGeometry(QtCore.QRect(90, 310, 91, 81))
        font = QtGui.QFont()
        font.setPointSize(15)
        self.btn_8.setFont(font)
        self.btn_8.setObjectName("btn_8")
        self.btn_9 = QtWidgets.QPushButton(self.centralwidget)
        self.btn_9.setGeometry(QtCore.QRect(180, 310, 91, 81))
        font = QtGui.QFont()
        font.setPointSize(15)
        self.btn_9.setFont(font)
        self.btn_9.setObjectName("btn_9")
        self.btn_0 = QtWidgets.QPushButton(self.centralwidget)
        self.btn_0.setGeometry(QtCore.QRect(0, 390, 91, 81))
        font = QtGui.QFont()
        font.setPointSize(15)
        self.btn_0.setFont(font)
        self.btn_0.setObjectName("btn_0")
        self.btn_point = QtWidgets.QPushButton(self.centralwidget)
        self.btn_point.setGeometry(QtCore.QRect(90, 390, 91, 81))
        font = QtGui.QFont()
        font.setPointSize(15)
        self.btn_point.setFont(font)
        self.btn_point.setObjectName("btn_point")
        self.btn_equally = QtWidgets.QPushButton(self.centralwidget)
        self.btn_equally.setGeometry(QtCore.QRect(180, 390, 91, 81))
        font = QtGui.QFont()
        font.setPointSize(15)
        self.btn_equally.setFont(font)
        self.btn_equally.setObjectName("btn_equally")
        self.btn_plus = QtWidgets.QPushButton(self.centralwidget)
        self.btn_plus.setGeometry(QtCore.QRect(270, 410, 61, 61))
        font = QtGui.QFont()
        font.setPointSize(15)
        self.btn_plus.setFont(font)
        self.btn_plus.setObjectName("btn_plus")
        self.btn_minus = QtWidgets.QPushButton(self.centralwidget)
        self.btn_minus.setGeometry(QtCore.QRect(270, 340, 61, 61))
        font = QtGui.QFont()
        font.setPointSize(15)
        self.btn_minus.setFont(font)
        self.btn_minus.setObjectName("btn_minus")
        self.btn_ymnoh = QtWidgets.QPushButton(self.centralwidget)
        self.btn_ymnoh.setGeometry(QtCore.QRect(270, 280, 61, 61))
        font = QtGui.QFont()
        font.setPointSize(15)
        self.btn_ymnoh.setFont(font)
        self.btn_ymnoh.setObjectName("btn_ymnoh")
        self.btn_del = QtWidgets.QPushButton(self.centralwidget)
        self.btn_del.setGeometry(QtCore.QRect(270, 220, 61, 61))
        font = QtGui.QFont()
        font.setPointSize(15)
        self.btn_del.setFont(font)
        self.btn_del.setObjectName("btn_del")
        self.btn_clear = QtWidgets.QPushButton(self.centralwidget)
        self.btn_clear.setGeometry(QtCore.QRect(270, 150, 61, 61))
        font = QtGui.QFont()
        font.setPointSize(15)
        self.btn_clear.setFont(font)
        self.btn_clear.setObjectName("btn_clear")
        MainWindow.setCentralWidget(self.centralwidget)

        self.retranslateUi(MainWindow)
        QtCore.QMetaObject.connectSlotsByName(MainWindow)

        self.add_func()

        self.is_equal = False

    def retranslateUi(self, MainWindow):
        _translate = QtCore.QCoreApplication.translate
        MainWindow.setWindowTitle(_translate("MainWindow", "MainWindow"))
        self.label.setText(_translate("MainWindow", "0"))
        self.btn_1.setText(_translate("MainWindow", "1"))
        self.btn_2.setText(_translate("MainWindow", "2"))
        self.btn_3.setText(_translate("MainWindow", "3"))
        self.btn_4.setText(_translate("MainWindow", "4"))
        self.btn_5.setText(_translate("MainWindow", "5"))
        self.btn_6.setText(_translate("MainWindow", "6"))
        self.btn_7.setText(_translate("MainWindow", "7"))
        self.btn_8.setText(_translate("MainWindow", "8"))
        self.btn_9.setText(_translate("MainWindow", "9"))
        self.btn_0.setText(_translate("MainWindow", "0"))
        self.btn_point.setText(_translate("MainWindow", "."))
        self.btn_equally.setText(_translate("MainWindow", "="))
        self.btn_plus.setText(_translate("MainWindow", "+"))
        self.btn_minus.setText(_translate("MainWindow", "-"))
        self.btn_ymnoh.setText(_translate("MainWindow", "*"))
        self.btn_del.setText(_translate("MainWindow", "/"))
        self.btn_clear.setText(_translate("MainWindow", "AC"))


    def add_func(self):
        self.btn_0.clicked.connect(lambda:self.write_number(self.btn_0.text()))
        self.btn_1.clicked.connect(lambda:self.write_number(self.btn_1.text()))
        self.btn_2.clicked.connect(lambda:self.write_number(self.btn_2.text()))
        self.btn_3.clicked.connect(lambda:self.write_number(self.btn_3.text()))
        self.btn_4.clicked.connect(lambda:self.write_number(self.btn_4.text()))
        self.btn_5.clicked.connect(lambda:self.write_number(self.btn_5.text()))
        self.btn_6.clicked.connect(lambda:self.write_number(self.btn_6.text()))
        self.btn_7.clicked.connect(lambda:self.write_number(self.btn_7.text()))
        self.btn_8.clicked.connect(lambda:self.write_number(self.btn_8.text()))
        self.btn_9.clicked.connect(lambda:self.write_number(self.btn_9.text()))
        self.btn_ymnoh.clicked.connect(lambda:self.write_number(self.btn_ymnoh.text()))
        self.btn_del.clicked.connect(lambda:self.write_number(self.btn_del.text()))
        self.btn_plus.clicked.connect(lambda:self.write_number(self.btn_plus.text()))
        self.btn_minus.clicked.connect(lambda:self.write_number(self.btn_minus.text()))
        self.btn_point.clicked.connect(lambda:self.write_number(self.btn_point.text()))

        self.btn_clear.clicked.connect(self.clear)

        self.btn_equally.clicked.connect(self.equally)

    def equally(self):
        self.label.setText(str(eval(self.label.text())))    
        self.is_equal = True  

    def clear(self):
        self.label.setText("0")

    def write_number(self, number):
        if self.label.text() == "0" or self.is_equal:
            self.label.setText(number)
            self.is_equal = False
        else:
            self.label.setText(self.label.text() + number)


if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    MainWindow = QtWidgets.QMainWindow()
    ui = Ui_Calc()
    ui.setupUi(MainWindow)
    MainWindow.show()
    sys.exit(app.exec_())